### PR TITLE
Make cel-java work in Graal native images

### PR DESCRIPTION
**Pull request by snazy**
The approach to make cel-java work in Graal native image files is to generate Graal native image meta files.

Introduces a Gradle plugin to scan the compiled classes and configuration-dependencies for classes that require an entry in `reflection-config.json`.

With the generated `reflection-config.json` + `native-image.properties` the Graal `native-image` tool has enough information to allow reflection for google-protobuf generated files.

Splits the `generated` project into `generated-pb`, which contains only the generated Java files for protobuf, and `generated-antlr`, which only contains the generated Java files for the CEL parser.

### CODE REVIEW

**Comment by snazy** [comment]
Note: includes a commit from #22

**Comment by nastra** [comment]
thanks for the ping. will try+review tomorrow

File: build-tools/src/main/kotlin/org/projectnessie/cel/tools/plugins/ReflectionConfigExtension.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.projectnessie.cel.tools.plugins
+
+import org.gradle.api.Project
```
Comment (nastra) [inline build-tools/src/main/kotlin/org/projectnessie/cel/tools/plugins/ReflectionConfigExtension.kt:19]: finally some Kotlin code, yay!

**Review by nastra — APPROVED** [review]
LGTM, thanks for fixing

**Pull request merged**